### ISSUE TITLE
[workspaces] allow storing individual files as 'recent workspaces'

### ISSUE THREAD

**Issue by vince-fugnitto** [theia-ide/theia#3887]
#### Proposal
Allow individual miscellaneous files to be included in the `Open Recent Workspaces...` command.

#### Reasoning
As a user, if I use Theia to edit a single file, I'd like to store the file as a recently opened item to easily return to it in the future. Vscode also permits this behaviour and lists the file as a recently opened item.

#### Implications
Since we would no longer be dealing with workspace folders or workspace files, we would probably also need to rename the menu and command item from `Open Recent Workspace...` to something more generic such as `Open Recent...`.

(example from `vscode`)
<img width="641" alt="screen shot 2018-12-21 at 9 42 31 am" src="https://user-images.githubusercontent.com/40359487/50348460-cd129300-0506-11e9-910f-f28fb8a81252.png">

**Comment by vince-fugnitto** [comment]
@lmcbout @elaihau 
I know you both did a lot of work implementing `Open Recent Workspaces...`, do you have any thoughts?

**Comment by elaihau** [comment]
If we do it, `Open recent workspaces` menu item needs to be renamed into `Open recent` (which is also the one used in vsCode).

It would be a nice feature, because vsCode does it - there should have been a discussion on reasons that the feature is helpful. 
from the functionality point of view however, it overlaps the `file search`, as users also see the recently opened file from the menu populated by CTRL+P.

**Comment by vince-fugnitto** [comment]
@elaihau yes I mentioned the menu / command would need to be updated to `Open Recent...`.
I'm not sure about the overlap with the `File Search` since that only occurs for a given workspace, not for some file I may have edited in the past.

**Comment by elaihau** [comment]
in `file search` the recent opened files are from `navigationLocationService.locations()`. I am not sure if files outside of the workspace are included or not.

**Comment by vince-fugnitto** [comment]
> in `file search` the recent opened files are from `navigationLocationService.locations()`. I am not sure if files outside of the workspace are included or not.

I tried and it only works if a workspace is currently opened. So in the case where we use Theia without a workspace to edit a file, we won't have the `File Search` at all.
